onnx/onnx.in.proto

```diff
@@ -497,6 +497,45 @@ message TensorShapeProto {
   repeated Dimension dim = 1;
 }
 
+// Sequences
+//
+// Defines a dense, ordered, collection of elements that are of homogeneous types.
+message SequenceProto {
+  // Sequences are made of TypeProto elements, which can either be tensors,
+  // maps, or sequences themselves.
+  message SequenceElement {
+    oneof value {
+      TensorProto tensor_value = 1;
+      SequenceProto sequence_value = 2;
+      MapProto map_value = 3;
+    };
+  };
+  repeated SequenceElement values = 1;
+}
+
+// Maps
+//
+// Specifies an associative table, defined by a key type and a value type.
+message MapProto {
+  // Map values are made of TypeProto elements, which can either be tensors,
+  // sequences, or maps themselves.
+  message MapElement {
+    oneof value {
+      TensorProto tensor_value = 1;
+      SequenceProto sequence_value = 2;
+      MapProto map_value = 3;
+    };
+  };
+
+  // Key-Value pair element
+  message KeyValuePair {
+    required int32 key = 1;
```

Comment:
The key needs to support [u]int types and string.

`TypeProto.Map.key_type` says:
```
    // This field MUST have a valid TensorProto.DataType value
    // This field MUST be present for this version of the IR.
    // This field MUST refer to an integral type ([U]INT{8|16|32|64}) or STRING
    optional int32 key_type = 1;
```

Reply:
Addressed in recent update, but unsure if this is the right way to do it.

Reply:
Rewritten as per to Bowen's suggestion of including datatype enum field and storing data in separate key fields similarly to TensorProto format in KeyValuePair.